src/main/java/parser/Parser.java

```diff
@@ -0,0 +1,244 @@
+            }
+
+            int index = Integer.parseInt(command.substring(5)) - 1;
+
+            if (index >= 0 && index < size) {
+                return new MarkCommand(index);
+            } else {
+                throw new TobiasException("Invalid number provided! Please give a valid index from 1 to " + size +" !!");
+            }
+        }
+        else if (command.startsWith("unmark")) {
+            int size = tasks.taskNum();
+
+            if (size == 0) {
+                throw new TobiasException("Your list is empty at the moment, add some todos/events/deadlines to UNMARK them!");
+            }
+
+            if (command.equals("unmark")) {
+                throw new TobiasException("No index provided! Please give a valid index from 1 to " + size +" !!");
+            }
+
+            String blank = command.substring(6,7);
+
+            if (!blank.isBlank()) {
+                throw new TobiasException("Kindly type your task a space after unmark!");
+            }
+
+            int index = Integer.parseInt(command.substring(7)) - 1;
+
+            if (index >= 0 && index < size) {
+                return new UnmarkCommand(index);
+            } else {
+                throw new TobiasException("Invalid number provided! Please give a valid index from 1 to " + size +" !!");
+            }
+        }
+        else if (command.startsWith("delete")) {
+            int size = tasks.taskNum();
+
+            if (size == 0) {
+                throw new TobiasException("Your list is empty at the moment, add some todos/events/deadlines to DELETE them!");
+            }
+
+            if (command.equals("delete")) {
+                throw new TobiasException("No index provided! Please give a valid index from 1 to " + size +" !!");
+            }
+
+            String blank = command.substring(6,7);
+
+            if (!blank.isBlank()) {
+                throw new TobiasException("Kindly type your task a space after delete!");
+            }
+
+            int index = Integer.parseInt(command.substring(7)) - 1;
+
+            if (index >= 0 && index < size) {
+                return new DeleteCommand(index);
+            } else {
+                throw new TobiasException("Invalid number provided! Please give a valid index from 1 to " + size +" !!");
+            }
+        }
+        else if (command.startsWith("find")) {
+            if (command.equals("find")) {
+                throw new TobiasException("Please enter what you are finding for after 'find' !!!");
+            }
+
+            String blank = command.substring(4,5);
+
+            if (!blank.isBlank()) {
+                throw new TobiasException("Kindly enter what you are finding a space after find !!!");
+            }
+
+            String keyWord = command.substring(5);
+            return new FindCommand(keyWord);
+        }
+        else if (command.equals("bye")) {
+            return new ExitCommand();
+        }
+        else {
```

Comment:
Again, maybe you would want to have else if and else statement on the same line as the closing brackets from the previous statement? This would follow our java coding standard:

```
if (condition) {
    statements;
} else if (condition) {
    statements;
} else {
    statements;
}
```